src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -647,20 +647,60 @@ private static TypeDeclDesc getTheNearestClass(String outerTypeDeclName,
      * @param secondTypeDecl second input type declaration
      * @return difference between type declaration name matching count
      */
-    private static int getTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
+    private static int getAnonTypeDeclarationNameMatchingCountDiff(String outerTypeDeclName,
```

Comment:
Similar to above, we are exposing implementation details in the method name, and it doesn't really help us to understand what this method does at a conceptual level or why we care about it.  Additionally, using `diff` and `match` in the name is misleading; once we are comparing depths, we really aren't talking about a "diff" any longer.

Please make this method signature to be `private static int calculateTypeDeclarationDistance(String outerTypeName, TypeDeclDesc firstType, TypeDeclDesc secondType)`.

Reply:
If we name it a bit more general, could it be problem of misusage? Non of arguments imply that implementation is designed for annon type only for now.
If someone wants to make general method, he can create with general name and do appreciate implementation.

Just think on this one more time, this had concerns with names even in original PR.
I don't want to let this PR stuck again. 
If you really see it should be changed , just confirm one more time, and we will do it.

Reply:
The existing name tells me everything except for what this method actually does 😂